networkx/algorithms/tree/mst.py

```diff
@@ -1125,19 +1125,26 @@ def _write_partition(self, partition):
             A Partition dataclass describing a partition on the edges of the
             graph.
         """
-        for u, v, d in self.G.edges(data=True):
-            if (u, v) in partition.partition_dict:
-                d[self.partition_key] = partition.partition_dict[(u, v)]
-            else:
-                d[self.partition_key] = EdgePartition.OPEN
+
+        partition_dict = partition.partition_dict
+        partition_key = self.partition_key
+        G = self.G
+
+        edges = G.edges(keys=True, data=True) if G.is_multigraph() else G.edges(data=True)
+        for *e, d in edges:
+            d[partition_key] = partition_dict.get(e, EdgePartition.OPEN)
 
     def _clear_partition(self, G):
         """
         Removes partition data from the graph
         """
-        for u, v, d in G.edges(data=True):
-            if self.partition_key in d:
-                del d[self.partition_key]
+        partition_key = self.partition_key
+        edges = (
+            G.edges(keys=True, data=True) if G.is_multigraph() else G.edges(data=True)
+        )
```

Comment:
While we're here, I think this looks cleaner. Feel free to disregard.

```suggestion
        edges = G.edges(keys=True, data=True) if G.is_multigraph() else G.edges(data=True)
```